File: src/dataset/c/clock_exercism-IrvinRamiA/gemini/GeminiFlash_clock_v1.c
#include "clock.h"
#include <stdio.h>
#include <stdbool.h>

clock_t clock_create(int hour, int minute) {
    clock_t clock;

    // Normalize hour and minute values
    hour = hour % 24;
    if (hour < 0) hour += 24; // Handle negative hours

    minute = minute % 60;
    if (minute < 0) minute += 60; // Handle negative minutes

    snprintf(clock.text, sizeof(clock.text), "%02d:%02d", hour, minute);

    return clock;
}

clock_t clock_add(clock_t clock, int minute_add) {
    int hour, minute;
    sscanf(clock.text, "%d:%d", &hour, &minute);

    minute += minute_add;
    hour += minute / 60;
    minute = minute % 60;

    if (minute < 0) {
        minute += 60;
        hour--;
    }

    hour = hour % 24;
    if (hour < 0) hour += 24;

    clock_t new_clock = clock_create(hour, minute);
    return new_clock;
}

clock_t clock_subtract(clock_t clock, int minute_subtract) {
    return clock_add(clock, -minute_subtract);
}

bool clock_is_equal(clock_t a, clock_t b) {
    return (strcmp(a.text, b.text) == 0);
}

File: src/dataset/python/clock_exercism-ThomasZumsteg/gemini/GeminiFlash_clock_v1.py
class Clock(object):
    def __init__(self, hour, minute):
        total_minutes = hour * 60 + minute
        total_minutes %= 1440  # 24 * 60
        self.hour = total_minutes // 60
        self.minute = total_minutes % 60

    def __repr__(self):
        return '{:02}:{:02}'.format(self.hour, self.minute)

    def __eq__(self, other):
        return self.hour == other.hour and self.minute == other.minute

    def __add__(self, minutes):
        return Clock(self.hour, self.minute + minutes)

    def __sub__(self, minutes):
        return Clock(self.hour, self.minute - minutes)